### PR TITLE
8949 - add email verification test

### PR DESCRIPTION
**Pull request by codyseibert**
- adding a cypress test to verify when a user changes their email, the user can successfully navigate to the verification link and get a success page once their email has been verified.
- refactoring the /auth/login endpoint to accept a mock user email address for the `?code=` which will generate a token on the backend and do the normal logic (set a cookie, return the access token).  The idea is we want to remove the various ways a user can login to our system and just have one approach.
- removing all instances where we save and load the token & user from localStorage.  These should be persisted in the cookie at this point so there is no need to store things in localStorage
- refactoring the pa11y tests to hit the /log-in?code endpoint instead of the previous /mock-login to reduce the amount of code paths

### CODE REVIEW

File: cypress-integration/integration/petitioner-updates-email.spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const {
+  confirmEmailVerificationSuccessful,
+} = require('../support/pages/public/email-verification');
+const { navigateTo: loginAs } = require('../support/pages/maintenance');
+
+describe('Petitioner updates and verifies their email', () => {
+  before(() => {
+    loginAs('petitioner9');
+    // wait because the UI flips views for some reason when you click on a button too soon when
+    // goto sequences are not Fully done (cerebral or riot router bug?)
```
Comment (rachelschneiderman) [inline cypress-integration/integration/petitioner-updates-email.spec.js:17]: This bug may be the same as what's reported in [this](https://trello.com/c/J7MLNXmT/824-route-sequence-refactor-options) devex card.

File: web-client/src/presenter/actions/setTokenAction.js [hunk 2]
```diff
@@ -3,15 +3,5 @@ import { state } from 'cerebral';
 export const setTokenAction = async ({ applicationContext, props, store }) => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/setTokenAction.js]: Nit: this action is no longer async

File: web-client/src/presenter/actions/setUserAction.js [hunk 3]
```diff
@@ -13,12 +13,4 @@ import { state } from 'cerebral';
 export const setUserAction = async ({ applicationContext, props, store }) => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/setUserAction.js]: Nit: remove async decorator

**Review by rachelschneiderman — APPROVED** [review]

File: cypress-integration/integration/petitioner-updates-email.spec.js (hunk 1 above)
Comment (codyseibert) [reply cypress-integration/integration/petitioner-updates-email.spec.js:17]: yeah it's probably related.  It's a strange behavior I've seen when trying to navigate too quickly when a sequence is already running.

**Pull request merged**